packages/core/tangleold/cmanaweightprovider.go

```diff
@@ -155,20 +157,25 @@ func (c *CManaWeightProvider) WeightsOfRelevantVoters() (weights map[identity.ID
 			return nil
 		})
 	}
-	c.clean(lowerBoundEpoch)
+	pruningPoint := c.confirmedEpochRetrieverFunc() - epoch.Index(activeEpochThreshold)
+	c.clean(pruningPoint)
 
 	return weights, totalWeight
 }
 
 // SnapshotEpochActivity returns the activity log for snapshotting.
-func (c *CManaWeightProvider) SnapshotEpochActivity() (epochActivity epoch.SnapshotEpochActivity) {
+func (c *CManaWeightProvider) SnapshotEpochActivity(epochDiffIndex epoch.Index) (epochActivity epoch.SnapshotEpochActivity) {
 	epochActivity = epoch.NewSnapshotEpochActivity()
 
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
 
 	for ei, al := range c.activityLog {
 		al.SetEpochs.ForEach(func(nodeID identity.ID) error {
```

Comment:
Instead of having ActivityLog exposing an AdvanceSet, can we create a ForEach on the activity record itself while unexporting the AdvancedSet itself?

Reply:
Now I use ForEach, but it turned out that I cannot unexport `SetEpochs` as the serialization stops working